File: LinkedList/725.cpp
class Solution {
public:
    vector<ListNode*> splitListToParts(ListNode* head, int k) {
        int total = 0;
        ListNode *prev = NULL, *curr = head;
        vector<ListNode*> res(k, nullptr);
        while (curr) {
            total++;
            curr = curr->next;
        }
        int size = total / k, remaining = total % k;
        curr = head;
        for (int i = 0; i < k; i++, --remaining) {
            res[i] = head;
            for (int j = 0; j < size + (remaining > 0); j++) {
                prev = head;
                head = head->next;
            }
            if (prev) prev->next = NULL;
        }
        return res;
    }
};

File: LinkedList/725.py
#725. Split Linked List in Parts
class Solution:
    def splitListToParts(self, head: Optional[ListNode], k: int) -> List[Optional[ListNode]]:
        prev, curr = None, head
        total = 0
        res = [None] * k
        while curr:
            total += 1
            curr = curr.next
        curr = head
        size, remaining = divmod(total, k)
    
        for i in range(k):
            res[i] = curr
            for j in range(size + int(remaining > 0)):
                prev = curr
                curr = curr.next
            if prev:
                prev.next = None
            remaining -= 1

        return res
